src/charm.py

```diff
@@ -51,53 +51,14 @@ def __init__(self, *args) -> None:
             ],
         )
 
-    def _require_nginx_route(self):
-        """Set require_nginx_route."""
-        service_hostname = self.config["external_hostname"] or self.app.name
-        service_name = self.app.name
-        service_port = GH_EXPORTER_WEBHOOK_PORT
-
-        if self._has_nginx_route_relation() and self._has_app_data():
-            rel = self.model.relations["nginx-route"][0].data[self.app]
-            if self._has_required_fields(rel):
-                service_hostname = self.config["external_hostname"] or self.app.name
-                service_name = rel[SVC_NAME]
-                service_port = rel[SVC_PORT]
-
-        require_nginx_route(
-            charm=self,
-            service_hostname=service_hostname,
-            service_name=service_name,
-            service_port=service_port,
-        )
-
-    def _has_required_fields(self, rel: Dict) -> bool:
-        """Check for required fields in relation.
-
-        Args:
-            rel: relation to check
-        Returns:
-            Returns true if all fields exist
-        """
-        return all(key in rel for key in (SVC_HOSTNAME, SVC_NAME, SVC_PORT))
-
-    def _has_app_data(self) -> bool:
-        """Check for app in relation data.
+    @property
+    def _github_actions_exporter(self) -> GitHubActionsExporter:
+        """Returns an instance of the GitHub Actions Exporter object.
 
         Returns:
-            Returns true if app data exist
+            Instance of GitHub Actions Exporter
         """
-        return self.app in self.model.relations["nginx-route"][0].data
-
-    def _has_nginx_route_relation(self) -> bool:
-        """Check for nginx-route relation.
-
-        Returns:
-            Returns true if nginx-route relation exist
-        """
-        return (
-            "nginx-route" in self.model.relations and len(self.model.relations["nginx-route"]) > 0
-        )
+        return GitHubActionsExporter(self)
```

Comment:
What is the reason for needing this please?

Reply:
Returns an instance of the class GitHubActionsExporter every time is needed to run the "/srv/gh_exporter/github-actions-exporter" command. In the case of this charm so far, we only need this to get the version of the exporter.

Reply:
Could that just be a function in the exporter module?

Reply:
Yes, changed. Thanks.